src/WpBookingsUiModule.php

```diff
@@ -123,18 +119,11 @@ public function run(ContainerInterface $c = null)
             $this->_adminMenu($c);
         });
 
-        $this->_attach('eddbk_bookings_visible_statuses', function ($event) use ($c) {
-            $event->setParams([
-                'statuses' => $this->_getVisibleStatuses($event->getParam('statuses'), $c->get('wp_bookings_ui/hidden_statuses')),
-            ]);
-        });
+        $this->_attach('eddbk_bookings_ui_state', $c->get('eddbk_bookings_ui_statuses_handler'));
 
-        $statusesOptionKey = $c->get('wp_bookings_ui/screen_options/key');
-        $this->_attach('wp_ajax_set_' . $statusesOptionKey, function () use ($statusesOptionKey) {
-            $data = json_decode(file_get_contents('php://input'), true);
-            $statuses = $data['statuses'];
-            $this->_setScreenStatuses($statusesOptionKey, $statuses);
-        });
+        $this->_attach('eddbk_bookings_ui_state', $c->get('eddbk_bookings_ui_status_transitions_handler'));
+
+        $c->get('eddbk_bookings_ui_statuses_handler')->run();
```

Comment:
What are you expecting here? Remember:

> When you are consuming a public method, you MUST be consuming an interface. The one exception is when the constructor is in the scope of the consuming code.

Reply:
Yes, I don't found interface for it. It would be good to discuss it in more details.

Reply:
As discussed in the call, you should split your fat statuses handler into thinner handlers with less responsibility, and attach them in the module.